crypto/engine/eng_cryptodev.c

```diff
@@ -1247,13 +1247,21 @@ cryptodev_asym(struct crypt_kop *kop, int rlen, BIGNUM *r, int slen,
         return (ret);
 
     if (r) {
-        kop->crk_param[kop->crk_iparams].crp_p = calloc(rlen, sizeof(char));
+        kop->crk_param[kop->crk_iparams].crp_p = OPENSSL_zalloc(rlen);
+        if (kop->crk_param[kop->crk_iparams].crp_p == NULL)
+            return (ret);
```

Comment:
No parens around "ret" in the return please (and may as well change other instances of this in this function while you are at it).